crates/wasi-common/yanix/src/dir.rs

```diff
@@ -78,18 +84,22 @@ impl Entry {
 
     /// Returns the type of this directory entry.
     pub fn file_type(&self) -> FileType {
-        unsafe { FileType::from_raw(self.0.d_type) }
+        FileType::from_raw(self.0.d_type)
     }
 }
 
 #[cfg(not(target_os = "android"))]
 #[derive(Clone, Copy, Debug)]
-pub struct SeekLoc(pub(crate) libc::c_long);
+pub struct SeekLoc(i64);
```

Comment:
If we're going to force `i64` here, we'll need to factor out  `Dir::seek` into separate implementations for emscripten and the rest since `libc::seekdir` in Emscripten expects seek loc as `i32`.

Reply:
Ok, I'll do this. I think this is the right way to go, so that wasi-common itself supports 64-bit offsets as far as it can.

Reply:
Oh, I didn't notice that seekdir/telldir/etc. actually are specified in terms of `long` in POSIX. That's surprising. For now, I'll just revert this SeekLoc change then.